cmd/acraserver/acraserver.go

```diff
@@ -178,8 +188,96 @@ func main() {
 			}
 		}()
 	}
-	if *withZone && !*disableHTTPApi {
-		go server.StartCommands()
+
+	sigHandlerSIGTERM, err := cmd.NewSignalHandler([]os.Signal{os.Interrupt, syscall.SIGTERM})
+	ErrorSignalChannel = sigHandlerSIGTERM.GetChannel()
+	if err != nil {
+		log.WithError(err).Errorln("can't register SIGTERM handler")
+		os.Exit(1)
+	}
+	go sigHandlerSIGTERM.Register()
+	sigHandlerSIGTERM.AddCallback(func() {
+		log.Infoln("Incoming SIGTERM or SIGINT")
+		// Stop accepting new connections
+		server.StopListeners()
+		// Wait a maximum of N seconds for existing connections to finish
+		err := server.WaitWithTimeout(ACRASERVER_WAIT_TIMEOUT * time.Second)
+		if err == errors.New("timeout") {
```

Comment:
better to compare with `WaitTimeoutError`

Reply:
done